### PR TITLE
 dotCMS Fields: create a contenttype fields library

### LINKED ISSUE #25781: dotCMS Fields: create a web components library
### Parent Issue

This possibility was explored in this PR: https://github.com/dotCMS/core/issues/25434

### Task

We are going to start to reimplement the new design of the edit `contentlet`, this means we need to create components for each field.

We would like to do web components for each field so we can add them to the JSP (old version) of the edit `contentlet` and offer the customer improvements as soon as we build it.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

Web components should be usable in `.jsp`

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

- When we build the library, the output should be a single `.js` file.

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by rjvelazco**
### Proposed Changes
* Create a library for contenttype fields

### <samp>🤖 Generated by Copilot at 4a2fc10</samp>

This pull request adds two new apps to the core-web project: contenttype-fields-builder and contenttype-fields-builder-e2e. The former is an Angular app that allows creating and editing custom contenttype fields components, and the latter is a Cypress app that provides end-to-end testing for the former. The pull request also updates the pre-commit hook script to exclude these apps from the linting target, and adds the necessary configuration files for ESLint, Cypress, Nx, Jest, and TypeScript for both apps.

## Related Issue
Fixes #25781

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 4a2fc10</samp>

*  Exclude new apps from linting target in pre-commit hook script ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-475b14c14bb44e7311cfc8e2df792c156a69f202bfe787513c985717eac57fa8L6-R6))
  - Add ESLint configuration file with Angular overrides ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-a50c7f87ee9a9cab84261c1d83b53d6aa191c21eb084ce75ca8c23fbec795d12R1-R36))
  - Add Jest configuration file with Angular and jest-preset-angular options ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-6b57e879582e3d9b6794ae178b1b1fcfc482c5854af88084e3cc832d0a6dc4b2R1-R22))
  - Add Nx project configuration file with build, serve, i18n, lint, and test targets ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-8ef1779598dab26e24d0460334ae8b17856ddc5466ebb2cd125f59bb93137cffR1-R104))
  - Add app component with app title ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-7da2635c2a92f3abaecd3ff3ca83a54e9b19d266cbad267a001d88ace7ec634dR1-R9))
  - Add app component test file with basic test cases ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-b0bc6144b4f40447e9380f2db21e4eb415e5828e4fcbc7f0f13ceef1833405cbR1-R32))
  - Add app module file that bootstraps app component and registers custom elements ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-07240e1adf9e007eb5e1201d172f486673b52f09fef6bd4383375b44b7ab3769R1-R42))
  - Add index HTML file with app root component ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-ef92bbc6d058cda69a2a5e04a42efbe3d058a822cefd45beb9af7a408fac003dR1-R13))
  - Add ESLint configuration file with Cypress overrides ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-885b517bb9b69f05df7ea5bfe1fc79e91ffe4ebdc3484505457397d332796042R1-R17))
  - Add Cypress configuration file with folder and runner options ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-727a9154e6a346b52920ab63fb67b88962e53db421763f8d444855fe2fe12fd3R1-R12))
  - Add Nx project configuration file with e2e and lint targets and dependency on contenttype-fields-builder app ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-1c620dac5f7e884a489414d8d816749e5b98cb8edab5ec251773771f0e8cf9aeR1-R31))
  - Add sample fixture file with mock data ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-f1f7b0aa07b7e3071c28afb887c7079fb560bc8505a5324b7ca49101ceb422d5R1-R4))
  - Add sample integration test file with simple test case ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-f8f7315f39c206e6d9f8033d5b5e9866a142fdcdd5cabdfe4dbd10740ab7f4a7R1-R13))
  - Add plugins file with TypeScript preprocessing ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-0cb75446e7e70bee53cdd9def788e6f3b5c3d4db82c6bd26c56bc6c6683c4e0aR1-R22))
  - Add helper function file to get greeting element ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-1181b65eb9cadf5d6d74e22c737c900e1fc969e98330b65f780cb90cf548b6ebR1))
  - Add custom commands file with login command ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-f5cb90fc8b87e50ff50d66944bf97228d9aef790fea114be5864487285099832R1-R33))
  - Add support file that imports custom commands file ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-91c01747fcf028cfae669988d14664890d7c67e5fe9adae762d198fd0efd9727R1-R17))
  - Add TypeScript configuration file with Cypress and Angular types and compiler options ([link](https://github.com/dotCMS/core/pull/25786/files?diff=unified&w=0#diff-b10e916522c07345c45cdd172e92a745300fb8c83573fd474888774de1f54068R1-R22))

### CODE REVIEW

File: core-web/apps/contenttype-fields-builder/src/app/app.component.spec.ts
Comment (manuelrojas) [inline core-web/apps/contenttype-fields-builder/src/app/app.component.spec.ts:1]: @rjvelazco it is looks like the Front-End checks are failing.

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/contenttype-fields/README.md
Comment (oidacra) [inline core-web/libs/contenttype-fields/README.md:1]: Update this file with what is this lib, how to run test, etc

File: core-web/apps/contenttype-fields-builder/src/app/app.component.spec.ts
Comment (rjvelazco) [reply core-web/apps/contenttype-fields-builder/src/app/app.component.spec.ts:1]: Good catch!

File: core-web/libs/contenttype-fields/README.md
Comment (rjvelazco) [reply core-web/libs/contenttype-fields/README.md:1]: I'll do it, thanks @oidacra!

File: core-web/apps/contenttype-fields-builder/src/main.ts
Comment (oidacra) [inline core-web/apps/contenttype-fields-builder/src/main.ts:1]: No standalone app?

**Review by manuelrojas — APPROVED** [review]

File: core-web/apps/contenttype-fields-builder/src/main.ts
Comment (rjvelazco) [reply core-web/apps/contenttype-fields-builder/src/main.ts:1]: This app has been specifically created for generating the `javascript` file, allowing the usage of `contenttype-fields` lib components in the `.jsp`.

Do you know a way to invoke `ngDoBootstrap` lifecycle method within `@Component()`?

File: core-web/apps/dotcms-ui/.storybook/main.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ module.exports = {
         '../src/**/*.stories.mdx',
         '../src/**/*.stories.@(js|jsx|ts|tsx)',
         '../../../libs/template-builder/**/*.stories.@(js|jsx|ts|tsx|mdx)',
-        '../../../libs/block-editor/**/*.stories.@(js|jsx|ts|tsx|mdx)'
+        '../../../libs/block-editor/**/*.stories.@(js|jsx|ts|tsx|mdx)',
+        '../../../libs/contenttype-fields/**/*.stories.@(js|jsx|ts|tsx|mdx)'
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/.storybook/main.js:15]: 😄

File: core-web/libs/contenttype-fields/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+## Running Unit Tests
+
+This lib uses `Jest` as unit test framework.
+
+- For run the contenttype-fields unit test do: `nx run contenttype-fields:test`.
+- For run the data-access unit test do: `nx run portlets-dot-experiments-data-access:test`.
+
+## Running lint
+
+- For run the portlet unit test do: `nx runnx run contenttype-fields:test:lint`.
```
Comment (fmontes) [inline core-web/libs/contenttype-fields/README.md:13]:
```suggestion
- For run the portlet unit test do: `nx run nx run contenttype-fields:test:lint`.
```

**Review by fmontes — CHANGES_REQUESTED** [review]

**Pull request merged**